layers/+lang/haskell/README.org

```diff
@@ -280,13 +318,14 @@ Refactor commands are prefixed by ~SPC m r~:
 Only some of the HLint suggestions can be applied.
 
 * Syntax checking
-At the moment there are three components, which can indicate
+There are multiple components that can indicate
```

Comment:
I am not sure I like this section. 

Basically it should list all available checkers, which are all for flycheck. This however should not list haskell-mode interactive as this is your repl, which may also have syntax checking in place but is not strictly working on your code files.

So my suggestion would be to remove haskell-mode interactive from this section as well as remove the (via .*) block.

Reply:
Well I have to admit I don't feel like I have enough knowledge to rewrite this section on my own, and I am not sure I 100% got what you are saying.

Are you proposing that we remove the first part completely, starting with "There are multiple" and ending with "- haskell-mode interactive"? And then that we also remove "** Interactive haskell-mode" section?

My thoughts:
1. I agree this section should be updated :D.
2. If we remove interactive-haskell mode section, are we losing any information? I don't see any other place mentioning error navigation shortcuts that are mentioned here (=haskell-goto-next-error= (~M-n~) and the other one), should we document those somewhere else? Under keybindings probably, but I am not sure if that should go under haskell mode or REPL hm. I could investigate that further, read more about haskell-mode, see if they are valid there also and so on.
3. If we remove that first section and interactive haskell section, not much is left in the Syntax checking section, right? Flycheck section talks about flycheck, which is something that could be looked up in syntax-checking layer docs. It mentions different checkers, but that feels more like implementation detail, why would use need to know explicitly about this? And finally it mentions HLint, but again it does not say anything that can't be looked up in Hlint documentation. So maybe just removing the whole section would make sense? Or making it much much shorter -> but what would you leave in it then?

Reply:
Good point, no idea yet, lets keep it as it is for now, we can later revise it again.

Reply:
Ok, just to confirm -> I left syntax checking section as it was, but removed interactive haskell.